sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -38,7 +38,7 @@ import org.apache.spark.sql.types.DataType
  * @since 1.3.0
  */
 @Stable
-sealed trait UserDefinedFunction {
+abstract class UserDefinedFunction {
```

Comment:
does scala have the concept of sealed abstract class?

Reply:
Good point, yes it does.